be/src/olap/task/index_builder.cpp

```diff
@@ -280,6 +278,11 @@ Status IndexBuilder::_write_inverted_index_data(TabletSchemaSPtr tablet_schema,
         auto column = tablet_schema->column(column_idx);
         auto writer_sign = std::make_pair(segment_idx, index_id);
         std::unique_ptr<Field> field(FieldFactory::create(column));
+        auto converted_result = _olap_data_convertor->convert_column_data(i);
```

Comment:
Is the problem avoid by using `_olap_data_convertor` after `if (column_idx < 0)` ?

Reply:
yes